### ISSUE TITLE
[Data] Add large-scale training data ingestion benchmarks to release tests

### ISSUE THREAD

**Issue by raulchen** [ray-project/ray#37355]
Currently we already have single-node training data ingestion benchmarks. 

To further harden the perf, robustness and user experience, we need to extend the current benchmark to be distributed and make sure it covers all common usages. 

Basic benchmark scenario:
1. Read training/validation data from S3.
1. Apply a CPU-based preprocessing map operation.
1. Feed into the Trainer and trigger distributed training on GPU nodes.

Things to test:
- Different file input types: s3 vs local disk, raw images vs parquet files.
- Different size of input files.
- Decompress data in preprocessing.
- Different levels of shuffling: randomize_block_order + different sizes of local shuffle buffer.
- Different number of epochs. 
- Different trainers: Torch/TF.
